go-controller/pkg/ovn/namespace.go

```diff
@@ -30,12 +31,24 @@ const (
 )
 
 func (oc *Controller) syncNamespaces(namespaces []interface{}) {
+	err := wait.PollImmediate(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		if err := oc.syncNamespacesRetriable(namespaces); err != nil {
+			klog.Errorf("Failed (will retry) in syncing namespaces: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		klog.Fatalf("Error in syncing namespaces: %v", err)
+	}
+}
+
+func (oc *Controller) syncNamespacesRetriable(namespaces []interface{}) error {
```

Comment:
good opportunity to add docstring?

Reply:
How about:
```
// This function implements the main body of work of syncNamespaces.
// Upon failure, it may be invoked multiple times in order to avoid a pod restart.
```